src/preset.ts

```diff
@@ -1470,37 +1503,44 @@ export async function expandBuildPreset(folder: string, name: string, workspaceF
     // Expand other fields
     if (preset.targets) {
         if (util.isString(preset.targets)) {
-            expandedPreset.targets = await expandString(preset.targets, expansionOpts);
+            expandedPreset.targets = await expandString(preset.targets, expansionOpts, errorHandler);
         } else {
             expandedPreset.targets = [];
             for (let index = 0; index < preset.targets.length; index++) {
-                expandedPreset.targets[index] = await expandString(preset.targets[index], expansionOpts);
+                expandedPreset.targets[index] = await expandString(preset.targets[index], expansionOpts, errorHandler);
             }
         }
     }
     if (preset.nativeToolOptions) {
         expandedPreset.nativeToolOptions = [];
         for (let index = 0; index < preset.nativeToolOptions.length; index++) {
-            expandedPreset.nativeToolOptions[index] = await expandString(preset.nativeToolOptions[index], expansionOpts);
+            expandedPreset.nativeToolOptions[index] = await expandString(preset.nativeToolOptions[index], expansionOpts, errorHandler);
         }
     }
 
     // Other fields can be copied by reference for simplicity
     merge(expandedPreset, preset);
 
+    if (errorHandler) {
+        for (const error of errorHandler.tempErrorList || []) {
```

Comment:
Why do we need to use a `tempErrorList`? Why not just use a single error list? It seems like we essentially already do this and then just assign it here?